tests/rgen/Microsoft.Macios.Bindings.Analyzer.Tests/NativeObjectHandleAnalyzerTests.cs

```diff
@@ -211,6 +211,19 @@ class Test
 					void Method(Class foo) { _ = foo.DangerousRetain().DangerousAutorelease().Handle; }
 				}
 				"""];
+			
+			// Equals implementation
+			yield return [
+				"""
+				using ObjCRuntime;
+
+				class Test : INativeObject
+				{
+					NativeHandle Handle { get; }
+					bool Equals(Test other) { return this.Handle == other.Handle; }
+				}
+				"""];
+			
 
 			// TODO: Test ThrowOnNull, GetConstant
```

Comment:
Note: This is the only non-addressed feedback in the last round. We will need to come up with some realistic test cases.